src/esssans/loki/masking.py

```diff
@@ -3,114 +3,58 @@
-) -> DetectorLowCountsStrawMask:
-    """
-    This mask aims to remove straws with low counts. It is based on the assumption
-    that the integrated counts in a straw should be at least greater than the number of
-    pixels in the straw (within a fudge factor of 0.5).
-    This is not true for straws that are not hit by the beam.
-    Because the straws are horizontal, this mask will typically mask straws that are
-    at the top and bottom edges of the detector panel.
-    Note that this has only been tested on the main rear detector panel.
-    """
-    return DetectorLowCountsStrawMask(
-        sample_straws.sum('pixel').data
-        < sc.scalar(sample_straws.sizes['pixel'] * 0.5, unit='counts')
-    )
-
-
-def detector_beam_stop_mask(
-    sample_straws: RawData[SampleRun],
-    beam_stop_position: BeamStopPosition,
-    beam_stop_radius: BeamStopRadius,
-) -> DetectorBeamStopMask:
-    """
-    This mask aims to remove the beam stop. It masks a circular region around the beam.
-    Usually, at this point in the workflow, the position of the beam is not accurately
-    know (the beam center finder has not been called yet, because it requires masked
-    data as an input). Therefore, only a rough estimate of the beam position is used.
-    Note that this has only been tested on the main rear detector panel.
-    """
-    pos = sample_straws.coords['position'] - beam_stop_position
-    pos.fields.z *= 0.0
-    return DetectorBeamStopMask((sc.norm(pos) < beam_stop_radius))
-
-
-def detector_tube_edge_mask(
-    sample_straws: RawData[SampleRun],
-) -> DetectorTubeEdgeMask:
-    """
-    This mask aims to remove the left and right edges of the detector panel.
-    Those areas typically have low counts. This mask is based on the assumption
-    that the counts integrated inside a vertical slab (summed over the ``straw`` and
-    ``layer`` dimensions) should be at least greater that the number of pixels that
-    make up the slab (within a fudge factor of 0.5).
-    This is not true for slabs that are not hit by the beam.
-    Note that this has only been tested on the main rear detector panel.
+    Parameters
+    ----------
+    data:
+        Raw data, defining the detector IDs.
+    masked:
+        The masked detector IDs.
     """
-    other_dims = set(sample_straws.dims) - {'pixel'}
-    size = np.prod([sample_straws.sizes[dim] for dim in other_dims])
-    return DetectorTubeEdgeMask(
-        sample_straws.sum(other_dims).data < sc.scalar(size * 0.5, unit='counts')
-    )
+    ids = data.coords['detector_number']
+    mask = sc.zeros(sizes=ids.sizes, dtype='bool')
+    mask.values[np.isin(ids.values, masked.values)] = True
+    return PixelMask(mask)
 
 
-def mask_detectors(
-    da: RawData[ScatteringRunType],
-    lowcounts_straw_mask: Optional[DetectorLowCountsStrawMask],
-    beam_stop_mask: Optional[DetectorBeamStopMask],
-    tube_edge_mask: Optional[DetectorTubeEdgeMask],
+def apply_pixel_masks(
```

Comment:
Why is this Loki-specific?

Reply:
Yes, this is another slightly annoying thing I had to work around.
In `isis`, the masks are applied on `RawDataWithComponentUserOffsets`, while in `loki`, they are applied on `RawData`.

Can we do it differently? I am not sure we want to introduce the `ComponentUserOffsets` everywhere?
I could make the `apply_pixel_masks` a common function and then have wrappers inside `isis` and `loki` that map the correct types, but I feel that would not be ideal either?

Reply:
What about Skadi etc.? My question was not necessarily if it can be the same for the ISIS instruments, more about all ESS instruments.

Reply:
Ok, will move into a common place for loki and leave the isis one in the submodule :+1: